lib/puppet/provider/rds_instance/v2.rb

```diff
@@ -0,0 +1,115 @@
+    config = {
+      ensure: :present,
+      name: instance.db_instance_identifier,
+      region: region,
+      engine: instance.engine,
+      engine_version: instance.engine_version,
+      db_instance_class: instance.db_instance_class,
+      master_username: instance.master_username,
+      db_name: instance.db_name,
+      allocated_storage: instance.allocated_storage,
+      storage_type: instance.storage_type,
+      license_model: instance.license_model,
+      multi_az: instance.multi_az,
+      iops: instance.iops,
+      db_parameter_group: instance.db_parameter_groups.collect(&:db_parameter_group_name).first,
+      db_security_groups: instance.db_security_groups.collect(&:db_security_group_name),
+    }
+    if instance.respond_to?('skip_final_snapshot')
+      config[:skip_final_snapshot] = instance.skip_final_snapshot
+    end
+    if instance.respond_to?('final_db_snapshot_identifier')
+      config[:final_db_snapshot_identifier] = instance.final_db_snapshot_identifier
+    end
+    if instance.respond_to?('backup_retention_period')
+      config[:backup_retention_period] = instance.backup_retention_period
+    end
+    if instance.respond_to?('endpoint') && !instance.endpoint.nil?
+      config[:endpoint] = instance.endpoint.address
+      config[:port]     = instance.endpoint.port
+    end
+    config
+  end
+
+  def exists?
+    dest_region = resource[:region] if resource
+    Puppet.info("Checking if instance #{name} exists in region #{dest_region || region}")
+    [:present, :creating, :available, :backing_up].include? @property_hash[:ensure]
+  end
+
+  def create
+    Puppet.info("Starting DB instance #{name}")
+    config = {
+      db_instance_identifier: resource[:name],
+      db_name: resource[:db_name],
+      db_instance_class: resource[:db_instance_class],
+      engine: resource[:engine],
+      engine_version: resource[:engine_version],
+      license_model: resource[:license_model],
+      storage_type: resource[:storage_type],
+      multi_az: resource[:multi_az].to_s,
+      allocated_storage: resource[:allocated_storage],
+      iops: resource[:iops],
+      master_username: resource[:master_username],
+      master_user_password: resource[:master_user_password],
+      subnet_group_name: resource[:db_subnet],
+      db_security_groups: resource[:db_security_groups],
+      db_parameter_group_name: resource[:db_parameter_group],
+    }
+
+    rds_client(resource[:region]).create_db_instance(config)
+
+    @property_hash[:ensure] = :present
+  end
+
+  def destroy
+    Puppet.info("Deleting database #{name} in region #{resource[:region]}")
+    rds = rds_client(resource[:region])
+    Puppet.info("Skip Final Snapshot: #{resource[:skip_final_snapshot]}")
+    config = {
+      db_instance_identifier: name,
+      skip_final_snapshot: resource[:skip_final_snapshot].to_s,
+      final_db_snapshot_identifier: resource[:final_db_snapshot_identifier],
+    }
+    rds.delete_db_instance(config)
```

Comment:
Good catch. On further investigation here these should be params not properties, they aren't set at creation and they aren't readable back from AWS. They only affect things at deletion time. I've pushed the changes and related test fixes. Cheers.